### ISSUE TITLE
Unable to specify keystoneURL in OpenStack CloudProfile

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#1441]
Steps to reproduce:
1. Apply CloudProfile.
```yaml
apiVersion: garden.sapcloud.io/v1beta1
kind: CloudProfile
metadata:
  name: os
spec:
  caBundle: | <caBundle>
  openstack:
    constraints:
      dnsProviders:
      - name: aws-route53
      - name: unmanaged
      floatingPools:
      - name: <name>
      loadBalancerProviders:
      - name: haproxy
      # [...]
    keystoneURL: https://<url>
```

2. Ensure that `.spec.openstack.keystoneURL`, `.spec.openstack.constraints.loadBalancerProviders` and `.spec.openstack.constraints.floatingPools` values are defaulted:
```bash
$ k get cloudprofile os -o yaml
```
```yaml
apiVersion: garden.sapcloud.io/v1beta1
kind: CloudProfile
metadata:
  name: os-eu-de-200
spec:
  openstack:
    constraints:
      dnsProviders:
      - name: aws-route53
      - name: unmanaged
      floatingPools: null
      loadBalancerProviders: null
      # [...]
    keystoneURL: ""
```

**Comment by ialidzhikov** [comment]
@rfranzke , are you able to reproduce this one?

**Comment by rfranzke** [comment]
Yes, very nice finding, thank you. I fixed it in [bfec3b9](https://github.com/gardener/gardener/pull/1430/commits/bfec3b9da85d192cac1cd3196dc305636537bfec) and added it to #1430. PTAL!
